include/deal.II/base/quadrature_lib.h

```diff
@@ -647,6 +647,206 @@ class QGaussLobattoChebyshev : public Quadrature<dim>
+ * Such transformation maps the reference square \$[0,1]\times[0,1]$ to the
+ * reference simplex, by collapsing the left \side of the square and squeezing
+ * quadrature points towards the orgin, and then shearing the resulting
+ * triangle to the reference one. This transformation shows good convergence
+ * properties when $\beta = 1$ with singularities of order $1/R$ in the origin,
+ * but different $\beta$ values can be selected to increase convergence and/or
+ * accuracy when higher order Gauss rules are used (see "Generalized Duffy
+ * transformation for integrating vertex singularities", S. E. Mousavi, N.
+ * Sukumar, Computational Mechanics 2009).
+ *
+ * When $\beta = 1$, this transformation is also known as the Lachat-Watson
+ * transformation.
+ *
+ * @author Luca Heltai, Nicola Giuliani, 2017.
+ */
+class  QDuffy: public QSimplex<2>
+{
+public:
+  /**
+   * Constructor that allows the specificatino of different quadrature rules
+   * along the "radial" and "angular" directions.
+   *
+   * Since this quadrature is not based on a Polar change of coordinates, it
+   * is not fully proper to talk about radial and angular directions. However,
+   * the effect of the Duffy transformation is similar to a polar change
+   * of coordinates, since the resulting quadrature points are aligned radially
+   * with respect to the singularity.
+   *
+   * @param radial_quadrature Base quadrature to use in the radial direction
+   * @param angular_quadrature Base quadrature to use in the angular direction
+   */
+  QDuffy(const Quadrature<1> &radial_quadrature,
+         const Quadrature<1> &angular_quadrature,
+         const double &beta = 1.0);
+
+  /**
+   * Calls the above constructor with QGauss<1>(n) quadrature formulas for
+   * both the radial and angular quadratures.
+   *
+   * @param n
+   */
+  QDuffy(const unsigned int &n, const double &beta);
+};
+
+/**
+ * A quadrature to use when the cell should be split in subregions to integrate
+ * using one or more base quadratures.
+ *
+ * @author Luca Heltai, 2017.
+ */
+template<int dim>
+class QSplit : public Quadrature<dim>
+{
+public:
+  /**
+   * Construct a quadrature formula by splitting the reference hyper cube into
+   * the minimum number of simplices that have vertex zero coinciding with
+   * `split_point`, and patch together affine transformations of the `base`
+   * quadrature.
+   *
+   * The resulting quadrature can be used, for example, to integrate functions
+   * with integrable singularities at the split point, provided that you select
+   * as base quadrature one that can integrate singular points on vertex zero
+   * of the reference simplex.
+   *
+   * An example usage in dimension two is given by:
+   * @code
+   * const unsigned int order = 5;
+   * QSplit<2> quad(QTrianglePolar(order), Point<2>(.3,.4));
+   * @endcode
+   *
+   * The resulting quadrature will look like the following:
+   * @image html split_quadrature.png ""
```

Comment:
So this ends up with 4 simplices. In 3d, the number of simplices is probably difficult to predict, or is that easy? Either way, what does that mean for the sum of weights that you expect in the `base` argument?

Reply:
Not really. There can be 2, 3, or 4 simplices according to where the point lies (vertex, edge or internal). For 3d, this is more complicated, as you noticed, but the principle is the same. According to where the split point is, you can have 6, 8, 10, or 12 tetrahedra (vertex, edge, face, or internal).

I've documented this.

Reply:
Oh, and by the way, if the quadrature is good for simplices, than QSplit is good for quads and hexes.

Reply:
(i.e., the sum of the weights in the end is always 1)

Reply:
How do you make sure the sum of weights is 1? By weighing the weights for each simplex by its area within the reference cell?

Reply:
The transformed simplex quadratures are obtained through `affine_transformation`, which also scales the weights according to `|det(B)|`. This ensures that each *transformed* simplex quadrature has weights that sum up to the area of the simplex. Notice that `QSimplex::affine_transformation` does not generate a `QSimplex`, but only a Quadrature, since the result is *not* on the unit simplex, but on an arbitrary simplex.

Reply:
Oh, yes. I recall.